src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1158,15 +1159,12 @@ void MarkEntireAssembly (AssemblyDefinition assembly)
 				MarkEntireType (type, includeBaseTypes: false, new DependencyInfo (DependencyKind.TypeInAssembly, assembly), null);
 		}
 
-		void ProcessModule (AssemblyDefinition assembly)
+		void ProcessModuleType (AssemblyDefinition assembly)
 		{
-			// Pre-mark <Module> if there is any methods as they need to be executed 
-			// at assembly load time
-			foreach (TypeDefinition type in assembly.MainModule.Types) {
-				if (type.Name == "<Module>" && type.HasMethods) {
-					MarkType (type, new DependencyInfo (DependencyKind.TypeInAssembly, assembly), null);
-					break;
-				}
+			// The <Modue> type may have an initializer, in which case we want to keep it.
+			TypeDefinition moduleType = assembly.MainModule.Types.Where (t => t.MetadataToken.RID == 1).Single ();
```

Comment:
Maybe `.FirstOrDefault()` and then check for null - to avoid hard crashing the linker if there's some really weird input (I know it should never happen, but I don't trust weird obfuscators and/or Cecil anymore :-))